cupy/core/core.pyx

```diff
@@ -1852,7 +1852,8 @@ cpdef function.Module compile_with_cache(
 
     return cuda.compile_with_cache(
         source, options, arch, cachd_dir, extra_source, backend,
-        enable_cooperative_groups=enable_cooperative_groups)
+        enable_cooperative_groups=enable_cooperative_groups,
+        specializations=specializations)
```

Comment:
`specializations` feels not good for [this use case](https://github.com/cupy/cupy/pull/3319#issuecomment-623054903). How about using `linkage` for example?

Reply:
Thanks, @asi1024. I have two questions:
1. I think it's more than the name `linkage` would imply. How about `name_expressions` following the NVRTC documentation?
2. Would you like this to be changed from all the way up (that is, the argument of RawModule)?

Reply:
1. `name_expressions` looks good :+1:
2. Yes, all `specializations` should be changed.

Reply:
LGTM